core/src/main/scala/com/yahoo/maha/utils/GetTotalRowsRequest.scala

```diff
@@ -0,0 +1,96 @@
+
+import scala.util.Try
+
+object GetTotalRowsRequest {
+  def updateRowList(rowList: RowList) : Unit = {
+    val row = rowList.newRow
+    println(rowList.columnNames)
+    rowList.columnNames.foreach {
+      col =>
+        row.addValue(col, s"$col-value")
+    }
+    rowList.addRow(row)
+  }
+
+  def getQueryExecutorContext: QueryExecutorContext = {
+    val qeOracle = new QueryExecutor {
+      override def execute[T <: RowList](query: Query, rowList: T, queryAttributes: QueryAttributes) : QueryResult[T] = {
+        updateRowList(rowList)
+        QueryResult(rowList, queryAttributes, QueryResultStatus.SUCCESS)
+      }
+      override def engine: Engine = OracleEngine
+    }
+
+    val qeHive = new QueryExecutor {
+      override def execute[T <: RowList](query: Query, rowList: T, queryAttributes: QueryAttributes) : QueryResult[T] = {
+        updateRowList(rowList)
+        QueryResult(rowList, queryAttributes, QueryResultStatus.SUCCESS)      }
+      override def engine: Engine = HiveEngine
+    }
+    val qeDruid = new QueryExecutor {
+      override def execute[T <: RowList](query: Query, rowList: T, queryAttributes: QueryAttributes) : QueryResult[T] = {
+        updateRowList(rowList)
+        QueryResult(rowList, queryAttributes, QueryResultStatus.SUCCESS)      }
+      override def engine: Engine = DruidEngine
+    }
+
+    val qec = new QueryExecutorContext
+    qec.register(qeOracle)
+    qec.register(qeHive)
+    qec.register(qeDruid)
+    qec
+  }
+
+  def getTotalRowsRequest(request: ReportingRequest, pipeline: QueryPipeline) : Try[ReportingRequest] = {
+    //no filters except fk filters
+    Try {
+      require(
+        pipeline.bestDimCandidates.nonEmpty
+        , s"Invalid total rows request, no best dim candidates! : $request")
+
+      //force dim driven
+      //remove all fields except primary key
+      //remove all sorts
+      val primaryKeyAliasFields = pipeline.bestDimCandidates.map(dim => Field(dim.publicDim.primaryKeyByAlias, None, None)).toIndexedSeq
+      request.copy(
+        selectFields = primaryKeyAliasFields
+        , sortBy = IndexedSeq.empty
+        , includeRowCount = true
+        , forceDimensionDriven = true
+        , forceFactDriven = false
+        , paginationStartIndex = 0
+        , rowsPerPage = request.rowsPerPage
+      )
+    }
+  }
+
+  def getTotalRows(request: ReportingRequest, sourcePipeline: QueryPipeline, registry: Registry) : Try[Int] = {
+    Try {
+      val totalRowsRequest: ReportingRequest = getTotalRowsRequest(request, sourcePipeline).get
+      val model: RequestModel = RequestModel.from(totalRowsRequest, registry).get
+      val maxRows: Int = DruidQueryGenerator.defaultMaximumMaxRows
+      assert(model.maxRows <= maxRows, throw new Exception(s"Value of ${model.maxRows} exceeds posted limit of $maxRows"))
+      val queryContext: QueryExecutorContext = getQueryExecutorContext
+
+      val rowListAttempt = sourcePipeline.execute(queryContext)
```

Comment:
This looks wrong, why would you want to execute the source pipeline?  Why aren't we creating a new pipeline from the new request model created from the total rows reporting request?